tools/crates-validator/src/lib.rs

```diff
@@ -466,87 +482,108 @@ impl Validator {
-        let res = if (status.success() && !profile.should_fail)
-            || (!status.success() && profile.should_fail)
+        let res = if (status.success() && !profile.settings.should_fail)
+            || (!status.success() && profile.settings.should_fail)
         {
-            info!("{} - run - SUCCESS", profile.name);
+            info!("{} - run - SUCCESS", profile.name());
             true
         } else {
             std::io::stdout().write_all(&output.stdout).unwrap();
             std::io::stderr().write_all(&output.stderr).unwrap();
 
-            error!("{} - run - FAILED", profile.name);
+            error!("{} - run - FAILED", profile.name());
             false
         };
 
         Ok(res)
     }
 
     /// Run a given profile through the set of tests
-    fn run(&self, profile: &mut CrateProfile) -> Result<()> {
-        if profiles::SKIP_CRATES.contains(&profile.name.as_str()) {
-            warn!("Skipping {} due to SKIP_CRATES", profile.name);
-            profile.results = Some(ValidationResults::from(RunStatus::Skipped));
-            return Ok(());
+    fn run(&self, profile: &Profile) -> Result<Vec<ValidationResults>> {
+        if self.context().skip_crates_names().contains(profile.name()) {
+            warn!("Skipping {}", profile.name());
+            return Ok(vec![ValidationResults::new(
+                profile.name(),
+                RunStatus::Skipped,
+                None,
+            )]);
         }
+
         let working_dir = self.gen_initial_project(profile)?;
-        self.customize_guest(profile, &working_dir)?;
-        let (build_success, build_errors) = self.build_project(profile, &working_dir)?;
-        if !build_success {
-            profile.results = Some(ValidationResults {
-                status: RunStatus::BuildFail,
-                build_errors: Some(build_errors),
-            });
-            return Ok(());
-        }
-        if profile.run_prover && !self.run_prover(profile, &working_dir)? {
-            profile.results = Some(ValidationResults::from(RunStatus::RunFail));
-            return Ok(());
+        let mut results = Vec::new();
+        for version in profile.versions.iter() {
+            self.customize_guest(profile, &version.clone(), &working_dir)?;
+            let (build_success, build_errors) = self.build_project(profile, &working_dir)?;
+            if !build_success {
+                results.push(ValidationResults::new(
+                    profile.name(),
+                    RunStatus::BuildFail,
+                    Some(build_errors),
+                ));
+                continue;
+            }
+            if profile.settings.run_prover && !self.run_prover(profile, &working_dir)? {
+                results.push(ValidationResults::new(
+                    profile.name(),
+                    RunStatus::RunFail,
+                    None,
+                ));
+                continue;
+            }
+            results.push(ValidationResults::new(
```

Comment:
With the `cargo risczero test` we might be able to some day run unit tests of a given crate (assuming we can clone the source). It might make sense to add a issue to track that as a future feature right here. We are currently in the process of running `risczero test` as part of our CI for our bitint acceleration forks so this might be a good additional follow on.

Reply:
Thanks for pointing out! 
Will be addressed in #958